plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/AlternateNumberFormat.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+
+module Ide.Plugin.AlternateNumberFormat where
+
+import           Control.Lens                         ((^.))
+import           Control.Monad.Except                 (ExceptT, MonadIO, liftIO)
+import qualified Data.HashMap.Strict                  as HashMap
+import           Data.Maybe                           (fromMaybe, isJust)
+import           Data.Text                            (Text)
+import qualified Data.Text                            as T
+import           Debug.Trace                          (traceM)
+import           Development.IDE                      (GetParsedModule (GetParsedModule),
+                                                       IdeState, RuleResult,
+                                                       Rules, define, ideLogger,
+                                                       realSrcSpanToRange,
+                                                       runAction, use,
+                                                       useWithStale)
+import           Development.IDE.Core.PositionMapping (PositionMapping)
+import           Development.IDE.GHC.Compat           hiding (getSrcSpan)
+import           Development.IDE.GHC.Compat.Util      (toList)
+import           Development.IDE.Graph.Classes        (Hashable, NFData)
+import           Development.IDE.Types.Logger         as Logger
+import           GHC.Generics                         (Generic)
+import           Ide.Plugin.Conversion                (FormatType,
+                                                       alternateFormat,
+                                                       toFormatTypes)
+import           Ide.Plugin.Literals                  (Literal (..),
+                                                       collectLiterals,
+                                                       getSrcSpan, getSrcText)
+import           Ide.Plugin.Retrie                    (handleMaybe,
+                                                       handleMaybeM, response)
```

Comment:
Acquiring a dependency on the retrie plugin just for these convenience functions seems like a bad idea. I would extract those functions to a shared place instead (or even duplicate them)

Reply:
A good place for them (without look at its implementaions) could be https://github.com/haskell/haskell-language-server/blob/master/hls-plugin-api/src/Ide/PluginUtils.hs

Reply:
Agreed. Would it be worth coming in with a follow up PR that replaces all usages? I pulled it from rename plugin. Looks like eval has it's own version in a util.

Reply:
So just to reconfirm. I will be making a separate PR to pull out the `handleMaybe` code for `eval`, `rename`, and `retrie` into a common area. So as of now this code will go. (unless that's not how you guys prefer things)